core/shared/src/main/scala/org/scalacheck/derive/MkShrink.scala

```diff
@@ -113,3 +114,37 @@ object MkCoproductShrink {
       }
     )
 }
+
+abstract class MkCoproductShrink0[C <: Coproduct] {
+  /** `Shrink[T]` instance built by this `MkCoproductShrink0[T]` */
+  def shrink: Shrink[C]
+  def singletons: Singletons[C]
+}
+
+object MkCoproductShrink0 {
+  def apply[T <: Coproduct](implicit mkShrink: MkCoproductShrink0[T]): MkCoproductShrink0[T] = mkShrink
+
+  def instance[T <: Coproduct](singletons0: Singletons[T])(shrink0: => Shrink[T]): MkCoproductShrink0[T] =
+    new MkCoproductShrink0[T] {
+      def shrink = shrink0
+      def singletons = singletons0
+    }
+
+  implicit val cnil: MkCoproductShrink0[CNil] =
+    instance(Singletons.empty)(Shrink.shrinkAny)
+
+  implicit def ccons[H, T <: Coproduct]
+   (implicit
+     headShrink: Strict[Shrink[H]],
+     tailShrink: MkCoproductShrink0[T],
+     headSingletons: Strict[Singletons[H]]
+   ): MkCoproductShrink0[H :+: T] =
+    instance[H :+: T](headSingletons.value.map(Inl(_)))(
```

Comment:
Shouldn't this include `++ tailShrink.singletons.map(Inr(_))`?

Reply:
Indeed!